pkg/channel/distributed/common/util/util.go

```diff
@@ -36,3 +38,23 @@ func WaitForSignal(logger *zap.Logger, signals ...os.Signal) {
 	// Log Signal Receipt
 	logger.Info("Received Signal", zap.String("Signal", sig.String()))
 }
+
+// Utility function to convert []byte headers to string ones for logging purposes
+func StringifyHeaders(headers []sarama.RecordHeader) map[string][]string {
+	stringHeaders := make(map[string][]string)
+	for _, header := range headers {
+		key := string(header.Key)
+		stringHeaders[key] = append(stringHeaders[key], string(header.Value))
+	}
+	return stringHeaders
+}
+
+// Pointer-version of the StringifyHeaders function
+func StringifyHeaderPtrs(headers []*sarama.RecordHeader) map[string][]string {
+	stringHeaders := make(map[string][]string)
+	for _, header := range headers {
+		key := string(header.Key)
+		stringHeaders[key] = append(stringHeaders[key], string(header.Value))
+	}
+	return stringHeaders
+}
```

Comment:
My kingdom for an interface!

should these be in `distributed/common/kafka/sarama` instead?  or maybe in the `eventing-kafka/pkg/common` ?

Reply:
kafka/sarama probably is a more relevant location.  We can move it out to the higher pkg/common if it becomes something we want in consolidated at that time.